pythran/pythonic/include/types/numpy_expr.hpp

```diff
@@ -107,7 +107,8 @@ namespace pythonic
       template <size_t I>
       bool _neq(numpy_expr_iterator const &other, utils::int_<I>) const
       {
-        return (std::get<I - 1>(iters_) != std::get<I - 1>(other.iters_)) ||
+        return (steps_[I - 1] &&
```

Comment:
If I correctly understant, thow iterator differ if they "(first don't have step (broadcast) and value differ) or another sub iterator differ"

In the case where there are no step (it broadcast), why you don't have to check that broadcasted value differ with  second one?